### PR TITLE
feat(boxai-sidebar): BoxAISidebar header added

### PR DESCRIPTION
**Pull request by DanilaRubleuski**
### Description
Added header to BoxAISidebar, which contains Expand button.

### Screenshots
image

### CODE REVIEW

**Comment by kkuliczkowski-box** [comment]
Because this is a public repo, we should not paste Jira tickets.

Shouldn't the "expand" button be right aligned?

File: src/elements/content-sidebar/BoxAISidebar.tsx [hunk 1]
```diff
@@ -24,16 +27,17 @@ export interface BoxAISidebarProps {
...
-function BoxAISidebar() {
+function BoxAISidebar({ onExpandPressed }: BoxAISidebarProps) {
     const { formatMessage } = useIntl();
 
     return (
         <SidebarContent
             className={'bcs-BoxAISidebar'}
             sidebarView={SIDEBAR_VIEW_BOXAI}
             title={formatMessage(messages.sidebarBoxAITitle)}
+            actions={<IconButton onClick={onExpandPressed} icon={ArrowsExpand} aria-label="Expand" size="x-small" />}
```
Comment (kkuliczkowski-box) [inline src/elements/content-sidebar/BoxAISidebar.tsx:38]: Should `aria-label` attribute be localized?

File: src/elements/content-sidebar/BoxAISidebar.tsx [hunk 2]
```diff
@@ -24,16 +27,17 @@ export interface BoxAISidebarProps {
...
         >
-            <div className="bcs-BoxAISidebar-content" />
+            <div className="bcs-BoxAISidebar-content"></div>
```
Comment (kkuliczkowski-box) [inline src/elements/content-sidebar/BoxAISidebar.tsx:40]: What is the reason for this change?

File: src/elements/content-sidebar/stories/BoxAISidebar.stories.tsx [hunk 3]
```diff
@@ -15,3 +13,5 @@ export default {
         token: global.TOKEN,
     },
 };
+
+export const basic = {};
```
Comment (kkuliczkowski-box) [inline src/elements/content-sidebar/stories/BoxAISidebar.stories.tsx:17]: What is the reason for moving this line below default export?

**Comment by MateuszMamczarz** [comment]
Looks like style of the header is not matching other headers. It should be bold, also not sure about font size

File: src/elements/content-sidebar/BoxAISidebar.tsx [hunk 4]
```diff
@@ -24,14 +27,22 @@ export interface BoxAISidebarProps {
     onExpandPressed: () => void;
 }
 
-function BoxAISidebar() {
+function BoxAISidebar({ onExpandPressed }: BoxAISidebarProps) {
```
Comment (jmalinna) [inline src/elements/content-sidebar/BoxAISidebar.tsx:30]: Could we change the name to `onExpandClicked`? Pressed doesn't sound natural to me 🤔Usually we name it as click

File: src/elements/content-sidebar/BoxAISidebar.scss [hunk 5]
```diff
@@ -0,0 +1,9 @@
+.bcs-BoxAISidebar {
+    .bcs-content-header {
+        justify-content: left;
+
+        .bcs-title {
+            padding: 0 12px 0 0;
```
Comment (jmalinna) [inline src/elements/content-sidebar/BoxAISidebar.scss:6]: Do we use Blueprint tokens in this repo and could replace 12px?

File: src/elements/content-sidebar/__tests__/BoxAISidebar.test.tsx [hunk 6]
```diff
@@ -16,4 +16,10 @@ describe('elements/content-sidebar/BoxAISidebar', () => {
 
         expect(screen.getByRole('heading', { level: 3, name: 'Box AI' })).toBeInTheDocument();
     });
+
+    test('should have accessible "Expand" button', () => {
+        renderComponent();
+
+        expect(screen.getByRole('button', { name: 'Expand' })).toBeInTheDocument();
```
Comment (jmalinna) [inline src/elements/content-sidebar/__tests__/BoxAISidebar.test.tsx:23]: Can we add `data-testid` to it and find element by this id so we won't need to update the test later on when the copy changes?
https://testing-library.com/docs/queries/bytestid/

File: src/elements/content-sidebar/BoxAISidebar.tsx (hunk 4 above)
Comment (jmalinna) [reply src/elements/content-sidebar/BoxAISidebar.tsx:30]: I think it's fine to have click in it `onExpandClicked` it indicates that it's a click handler

File: src/elements/content-sidebar/__tests__/BoxAISidebar.test.tsx (hunk 6 above)
Comment (jmalinna) [reply src/elements/content-sidebar/__tests__/BoxAISidebar.test.tsx:23]: I mean it's a question to @jankowiakdawid

File: src/elements/common/messages.js [hunk 7]
```diff
@@ -532,6 +532,11 @@ const messages = defineMessages({
         description: 'Label for transcript skill section in the preview sidebar',
         defaultMessage: 'Transcript',
     },
+    expandBoxAI: {
```
Comment (wpiesiak) [inline src/elements/common/messages.js:535]: I feel like it should be moved to `src/elements/content-sidebar/messages.js`

File: src/elements/content-sidebar/BoxAISidebar.scss [hunk 8]
```diff
@@ -0,0 +1,11 @@
+@import '~@box/blueprint-web-assets/tokens/tokens.scss';
+
+.bcs-BoxAISidebar {
+    .bcs-content-header {
+        justify-content: left;
+
+        .bcs-title {
+            padding: 0 $space-3 0 0;
```
Comment (jankowiakdawid) [inline src/elements/content-sidebar/BoxAISidebar.scss:8]: Isn't `padding-top` enough? Do wee need to clear other sides?

File: src/elements/content-sidebar/stories/tests/BoxAISidebar-visual.stories.tsx [hunk 9]
```diff
@@ -24,3 +24,12 @@ export const BoxAIInSidebar: StoryObj<typeof BoxAISidebar> = {
         expect(sidebar).toBeInTheDocument();
     },
 };
+
+export const ExpandButtonCheck: StoryObj<typeof BoxAISidebar> = {
+    play: async ({ canvasElement }) => {
+        const canvas = within(canvasElement);
+
+        const expandButton = await canvas.findByRole('button', { name: 'Expand' }, { timeout: 5000 });
+        await userEvent.click(expandButton);
```
Comment (jankowiakdawid) [inline src/elements/content-sidebar/stories/tests/BoxAISidebar-visual.stories.tsx:33]: At the moment the this tests only checks if button can be clicked. But not if anything happens. I think we should add at least an `expect` if callback function was called.

File: src/elements/content-sidebar/__tests__/BoxAISidebar.test.tsx (hunk 6 above)
Comment (jankowiakdawid) [reply src/elements/content-sidebar/__tests__/BoxAISidebar.test.tsx]: Check out https://testing-library.com/docs/queries/about#priority

tl;dr
there are three types of queries
1. based on accessibility - ie `*byRole` `*byText` `*byDisplayValue`
2. based on semantics - `*byAltText`, `*byTitle`
3. based on testid - `*byTestId`

It's always preferable to use accessible queries as they also check if our UI is accessible.
`data-testid` is an escape hatch in rare cases where it's inconvenient to find element of interest with accessible/semantic queries.

Comment (jmalinna) [reply src/elements/content-sidebar/__tests__/BoxAISidebar.test.tsx]: Thanks Dawid!

File: src/elements/content-sidebar/BoxAISidebar.scss [hunk 10]
```diff
@@ -0,0 +1,11 @@
+@import '~@box/blueprint-web-assets/tokens/tokens.scss';
+
+.bcs-BoxAISidebar {
+    .bcs-content-header {
+        justify-content: left;
```
Comment (greg-in-a-box) [inline src/elements/content-sidebar/BoxAISidebar.scss:5]: Are we certain we wanna push it to the left instead of the right?, the other tabs have buttons coming from right to left instead of left to right

**Review by jmalinna — APPROVED** [review]
LGTM

File: src/elements/content-sidebar/BoxAISidebar.scss (hunk 10 above)
Comment (MateuszMamczarz) [reply src/elements/content-sidebar/BoxAISidebar.scss:5]: There will be more controls added to this header, that's why it's aligned to left right now.

**Review by greg-in-a-box — DISMISSED** [review]

File: src/elements/content-sidebar/BoxAISidebar.scss (hunk 10 above)
Comment (JChan106) [reply src/elements/content-sidebar/BoxAISidebar.scss:5]: Even if we don't have all the controls currently, can we still have the proper formatting in place so we don't miss it? We are looking to have the expand button on the right side right? So something like a `space-between`?

File: src/elements/content-sidebar/BoxAISidebar.scss [hunk 11]
```diff
@@ -0,0 +1,11 @@
...
+
+        .bcs-title {
+            padding-right: $space-3;
```
Comment (JChan106) [inline src/elements/content-sidebar/BoxAISidebar.scss:8]: Could probably use gap in this case?

File: src/elements/content-sidebar/messages.js [hunk 12]
```diff
@@ -117,6 +117,11 @@ const messages = defineMessages({
         defaultMessage: 'Modify General Task',
         description: 'modal title for when editing an existing general task',
     },
+    expandBoxAI: {
+        id: 'be.expandBoxAI',
```
Comment (tjuanitas) [inline src/elements/content-sidebar/messages.js:121]: we should add `be.contentSidebar` to the id prefix since we dont want to risk any conflicts

File: src/elements/content-sidebar/stories/tests/BoxAISidebar-visual.stories.tsx [hunk 13]
```diff
@@ -24,3 +24,12 @@ export const BoxAIInSidebar: StoryObj<typeof BoxAISidebar> = {
         expect(sidebar).toBeInTheDocument();
     },
 };
+
+export const ExpandButtonCheck: StoryObj<typeof BoxAISidebar> = {
```
Comment (tjuanitas) [inline src/elements/content-sidebar/stories/tests/BoxAISidebar-visual.stories.tsx:28]: i assume this test will become a visual regression for the entire expanded view in Box AI right? so maybe we can name it like `BoxAIWithExpandedView`

File: src/elements/content-sidebar/stories/BoxAISidebar.stories.tsx (hunk 3 above)
Comment (tjuanitas) [reply src/elements/content-sidebar/stories/BoxAISidebar.stories.tsx:17]: just to reiterate, the `default` export is usually on the bottom, so the `basic` export should be moved above it

File: src/elements/content-sidebar/__tests__/BoxAISidebar.test.tsx [hunk 14]
```diff
@@ -16,4 +19,19 @@ describe('elements/content-sidebar/BoxAISidebar', () => {
...
+    });
+
+    test('should call onExpandClick when click "Expand" button', async () => {
+        renderComponent();
+
+        const expandButton = screen.getByRole('button', { name: 'Expand' });
+        await userEvent.click(expandButton);
+
+        expect(mockOnExpandClick).toHaveBeenCalled();
```
Comment (tjuanitas) [inline src/elements/content-sidebar/__tests__/BoxAISidebar.test.tsx:35]: there can be issues and unexpected behavior if we forget to clear/reset the mock calls between tests. i think we handle this in a global configuration but just in case, it's better to not use reuse the fn mock for the `expect` so it's more intentional about what should happen

e.g. check for a separate prop (defined in the local scope) and/or explicitly add an `afterEach`
```
afterEach(() => {
    jest.clearAllMocks();
});

// or

const onExpandClick = jest.fn();
renderComponent({ onExpandClick });

...

expect(onExpandClick).toHaveBeenCalled();
```

File: src/elements/content-sidebar/messages.js [hunk 15]
```diff
@@ -117,6 +117,11 @@ const messages = defineMessages({
         defaultMessage: 'Modify General Task',
         description: 'modal title for when editing an existing general task',
     },
+    expandBoxAI: {
+        id: 'be.contentSidebarExpandBoxAI',
```
Comment (jstoffan) [inline src/elements/content-sidebar/messages.js:121]: Can we namespace this to be similar to the other messages? Maybe `be.contentSidebar.boxAI.expand`?

File: src/elements/content-sidebar/messages.js [hunk 16]
```diff
@@ -117,6 +117,11 @@ const messages = defineMessages({
...
+        description: 'Default message for Box AI expand button in sidebar header',
+        defaultMessage: 'Expand',
```
Comment (jstoffan) [inline src/elements/content-sidebar/messages.js:123]: Minor, but let's move this up a line to stay consistent with the other messages.

File: src/elements/content-sidebar/messages.js [hunk 17]
```diff
@@ -117,6 +117,11 @@ const messages = defineMessages({
         defaultMessage: 'Modify General Task',
         description: 'modal title for when editing an existing general task',
     },
+    expandBoxAI: {
```
Comment (jstoffan) [inline src/elements/content-sidebar/messages.js:120]: `boxAISidebarExpand` seems accurate?

File: src/elements/content-sidebar/BoxAISidebar.tsx [hunk 18]
```diff
@@ -15,23 +17,32 @@ import { EVENT_JS_READY } from '../common/logger/constants';
...
+    onExpandClick: () => void;
 }
 
-function BoxAISidebar() {
+function BoxAISidebar({ onExpandClick }: BoxAISidebarProps) {
     const { formatMessage } = useIntl();
 
     return (
         <SidebarContent
             className={'bcs-BoxAISidebar'}
```
Comment (jstoffan) [inline src/elements/content-sidebar/BoxAISidebar.tsx:35]: Minor, but this can be a regular string: `className="bcs-BoxAISidebar"`

File: src/elements/content-sidebar/BoxAISidebar.tsx [hunk 19]
```diff
@@ -15,23 +17,32 @@ import { EVENT_JS_READY } from '../common/logger/constants';
...
             sidebarView={SIDEBAR_VIEW_BOXAI}
             title={formatMessage(messages.sidebarBoxAITitle)}
+            actions={
+                <IconButton
+                    onClick={onExpandClick}
+                    icon={ArrowsExpand}
+                    aria-label={formatMessage(sidebarMessages.expandBoxAI)}
```
Comment (jstoffan) [inline src/elements/content-sidebar/BoxAISidebar.tsx:42]: We typically alphabetize our props whenever possible.

**Review by jstoffan — COMMENTED** [review]
LGTM apart from some minor feedback.

**Review by jstoffan — APPROVED** [review]

**Review by jstoffan — APPROVED** [review]

**Review by tjuanitas — APPROVED** [review]

**Review by jstoffan — APPROVED** [review]

**Review by tjuanitas — APPROVED** [review]
